File: recipes/ascat-simulated-tumors-hg19/scripts/prepare.R
#!/usr/bin/env Rscript
# SPDX-License-Identifier: CC0-1.0

# Run the complete ASCAT example-data workflow from recipe-relative paths.

file_arg <- grep("^--file=", commandArgs(trailingOnly = FALSE), value = TRUE)
if (length(file_arg) != 1) {
    stop("Run this script using Rscript.", call. = FALSE)
}

script_path <- normalizePath(sub("^--file=", "", file_arg), mustWork = TRUE)
scripts_dir <- dirname(script_path)
recipe_dir <- dirname(scripts_dir)

Sys.setenv(
    ASCAT_INPUT_DIR = file.path(recipe_dir, "download"),
    ASCAT_WORK_DIR = file.path(recipe_dir, "work"),
    ASCAT_DATA_DIR = file.path(recipe_dir, "work"),
    ASCAT_OUTPUT_DIR = file.path(recipe_dir, "output")
)

run_script <- function(filename) {
    message("Running ", filename)
    status <- system2(
        file.path(R.home("bin"), "Rscript"),
        file.path(scripts_dir, filename)
    )
    if (status != 0) {
        stop(filename, " failed with status ", status, call. = FALSE)
    }
}

run_script("prepare-objects.R")
run_script("wrangle.R")
run_script("validate.R")

if (Sys.getenv("ASCAT_SOURCE_DIR") != "") {
    run_script("compare-distances.R")
}

File: recipes/_template/scripts/prepare.py
#!/usr/bin/env python3
# SPDX-License-Identifier: CC0-1.0

# /// script
# requires-python = ">=3.12"
# dependencies = []
# ///

"""Replace with a deterministic preparation workflow."""

from pathlib import Path


def main() -> None:
    """Create standard working directories and stop at the template boundary."""

    recipe_dir = Path(__file__).resolve().parents[1]
    for name in ("download", "work", "output"):
        (recipe_dir / name).mkdir(exist_ok=True)
    raise NotImplementedError("Replace the recipe template implementation.")


if __name__ == "__main__":
    main()
